src/Libraries/Microsoft.AspNetCore.Diagnostics.Middleware/Logging/LoggingRedactionOptions.cs

```diff
@@ -0,0 +1,88 @@
+    /// Default set to <see cref="IncomingPathLoggingMode.Formatted"/>.
+    /// This option only applies when the <see cref="RequestPathParameterRedactionMode"/>
+    /// option is not set to <see cref="HttpRouteParameterRedactionMode.None"/>.
+    /// </remarks>
+    public IncomingPathLoggingMode RequestPathLoggingMode { get; set; } = DefaultRequestPathLoggingMode;
+
+    /// <summary>
+    /// Gets or sets a value indicating how request path parameter should be redacted.
+    /// </summary>
+    /// <remarks>
+    /// Default set to <see cref="HttpRouteParameterRedactionMode.Strict"/>.
+    /// </remarks>
+    public HttpRouteParameterRedactionMode RequestPathParameterRedactionMode { get; set; } = DefaultPathParameterRedactionMode;
+
+    /// <summary>
+    /// Gets a map between HTTP path parameters and their data classification.
+    /// </summary>
+    /// <remarks>
+    /// Default set to an empty dictionary.
+    /// If a parameter within a controller's action is not annotated with a data classification attribute and
+    /// it's not found in this map, it will be redacted as if it was <see cref="DataClassification.Unknown"/>.
+    /// If you don't want a parameter to be redacted, mark it as <see cref="DataClassification.None"/>.
+    /// </remarks>
+    public IDictionary<string, DataClassification> RouteParameterDataClasses { get; } = new Dictionary<string, DataClassification>(StringComparer.OrdinalIgnoreCase);
+
+    /// <summary>
+    /// Gets a map between request headers to be logged and their data classification.
+    /// </summary>
+    /// <remarks>
+    /// Default set to an empty dictionary.
+    /// That means that no request header will be logged by default.
+    /// </remarks>
+    public IDictionary<string, DataClassification> RequestHeadersDataClasses { get; } = new Dictionary<string, DataClassification>(StringComparer.OrdinalIgnoreCase);
+
+    /// <summary>
+    /// Gets a map between response headers to be logged and their data classification.
+    /// </summary>
+    /// <remarks>
+    /// Default set to an empty dictionary.
+    /// That means that no response header will be logged by default.
+    /// </remarks>
+    public IDictionary<string, DataClassification> ResponseHeadersDataClasses { get; } = new Dictionary<string, DataClassification>(StringComparer.OrdinalIgnoreCase);
+
+    /// <summary>
+    /// Gets the set of HTTP paths that should be excluded from logging.
+    /// </summary>
+    /// <remarks>
+    /// Any path added to the set will not be logged.
+    /// Paths are case insensitive.
+    /// Default set to an empty <see cref="HashSet{T}"/>.
+    /// </remarks>
+    /// <example>
+    /// A typical set of HTTP paths would be:
+    /// - "/probe/live".
+    /// - "/probe/ready".
+    /// </example>
+    public ISet<string> ExcludePathStartsWith { get; } = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
```

Comment:
Is this redundant with the new features like short circuit routes, logging attributes, etc.?

Reply:
Not really, some folks implement their service's readiness/liveness probes in a controller and there's no point of logging these requests.
Are there logging attributes they can apply on a controller/action to exclude it from being logged?

Reply:
We can still decide to keep the option as it provides the ability to change the behavior without recompiling the code. And some folks might not have an access to the source code of some controllers.
@geeknoid @noahfalk thoughts?